File: DocMgmtApi/DocMgmtApi/Models/Document.cs
namespace DocMgmtApi.Models
{
    public class Document
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public byte[] Data { get; set; }

        public DateTime DateCreated { get; set; } = DateTime.Now;
        public Guid CustomerId { get; set; } // Required foreign key property
    }
}

File: DocMgmtUi/docmgmtui/src/types/Document.ts


export type Document = {
    id?: string;
    name: string;
    data: string;
    type: string;
    dateCreated: string;
    customerId: string;
}

export const emptyDocument = {
    name: '',
    data: '',
    type: 'Driving License',
    dateCreated: '',
    customerId: '',
}
